### PR TITLE
BMD-7309: send notifications to SNS instead of SQS

### PR DESCRIPTION
**Pull request by bmxpapa**
before this PR, infra with SNS for production need to be reopened & applied https://github.com/BerlingskeMedia/terraform/pull/443

### CODE REVIEW

**Review by bmxmaza — APPROVED** [review]

**Review by bmxpiku — CHANGES_REQUESTED** [review]

File: server/kafka.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ if(!KAFKA_BROKERS) {
   console.log('Kafka disabled')
   module.exports = new EventEmitter();
   module.exports.disabled = true;
+  module.exports.healthcheck = () => true;
```
[the remark this answers is not in the document]
Comment (bmxpapa) [reply server/kafka.js:19]: previously in that case application was throw with a function doesnt exists, not sure if false or void is a better option (then we need to change the place where healthcheck is checked)

File: server/sns.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+// https://aws.amazon.com/about-aws/whats-new/2013/06/18/amazon-sqs-announces-256KB-large-payloads/
+const SNS_TOPIC_MAX_SIZE = parseInt(process.env.SNS_TOPIC_MAX_SIZE || 256000);
+
+AWS.config.update({ accessKeyId: AWS_ACCESS_KEY_ID, secretAccessKey: AWS_SECRET_ACCESS_KEY, region: AWS_REGION });
+
+if (AWS_ALLOW_SELF_CERT) {
+  AWS.config.update({ httpOptions: { agent: new https.Agent({ rejectUnauthorized: false }) }});
+}
+
+const sns = new AWS.SNS({ apiVersion: '2010-03-31', endpoint: SNS_ENDPOINT } );
```
[the remark this answers is not in the document]
Comment (bmxpapa) [reply server/sns.js:28]: it is an API version

File: server/sns.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+const AWS_REGION = process.env.AWS_REGION || 'eu-west-1';
+const AWS_ACCESS_KEY_ID = process.env.AWS_ACCESS_KEY_ID;
+const AWS_SECRET_ACCESS_KEY = process.env.AWS_SECRET_ACCESS_KEY;
+const SNS_ENDPOINT = process.env.SNS_ENDPOINT;
+const AWS_ALLOW_SELF_CERT = process.env.AWS_ALLOW_SELF_CERT;
+const SNS_TOPIC = process.env.SNS_TOPIC || 'AriaDataDispatcherNotifications';
+// https://aws.amazon.com/about-aws/whats-new/2013/06/18/amazon-sqs-announces-256KB-large-payloads/
+const SNS_TOPIC_MAX_SIZE = parseInt(process.env.SNS_TOPIC_MAX_SIZE || 256000);
+
+AWS.config.update({ accessKeyId: AWS_ACCESS_KEY_ID, secretAccessKey: AWS_SECRET_ACCESS_KEY, region: AWS_REGION });
```
[the remark this answers is not in the document]
Comment (bmxpapa) [reply server/sns.js:22]: https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-environment.html

File: server/notifications_events.js [hunk 4]
```diff
@@ -103,22 +96,19 @@ module.exports = {
           event_id = "4";
 
         }
-        if(!CONSOLE_LOG_EVENTS) {
-          console.info(JSON.stringify({event_id}));
+        if (!CONSOLE_LOG_EVENTS) {
+          console.info('event_id:', JSON.stringify(event_id));
```
[the remark this answers is not in the document]
Comment (bmxpapa) [reply server/notifications_events.js:100]: It was for debug purpose for me: more descriptive,  I will remove it

File: server/kafka.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bmxpapa) [reply server/kafka.js]: nvm, fixed to false

File: server/sns.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (bmxpiku) [reply server/sns.js:23]: I mean :D I get that, although maybe this should go via some env var? so its easier to upgrade?

**Review by bmxpiku — APPROVED** [review]

**Review by bmxmaza — APPROVED** [review]

**Pull request merged**